sys/include/net/gnrc/gnrc_mac/gnrc_mac_types.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_mac  A common MAC type for providing key MAC parameters and helper functions
+ * @ingroup     net
+ * @brief       Internal types of GNRC_MAC
+ * @{
+ *
+ * @file
+ * @brief       Internal types used by the GNRC_MAC entities
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_MAC_TYPES_H_
+#define GNRC_MAC_TYPES_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <kernel_types.h>
+#include <xtimer.h>
+#include <net/netdev2.h>
+#include <net/gnrc/netdev2.h>
+#include <net/gnrc.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/******************************************************************************/
+
+typedef enum {
+    TX_FEEDBACK_UNDEF = -1,
+    TX_FEEDBACK_SUCCESS,
+    TX_FEEDBACK_NOACK,
+    TX_FEEDBACK_BUSY
+} gnrc_mac_tx_feedback_t;
+
+/******************************************************************************/
+typedef struct gnrc_mac {
+    /* PID of MAC thread */
+    kernel_pid_t pid;
+    /* NETDEV device used by MAC */
+    gnrc_netdev2_t* netdev;
```

Comment:
At least these to above are already provided by [`gnrc_netdev2_t`](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/gnrc/netdev2.h#L75-L83), which should be used to implement MAC anyways by providing a `send` and `recv` implementation in my opinion. I also think that was how @kaspar030 intended it. Feel free to amend to that struct, but I think it could be beneficial to `#ifdef MODULE_GNRC_MAC` them out